odoo/addons/base/models/ir_ui_view.py

```diff
@@ -2016,7 +2016,11 @@ def __init__(self, validate, Model):
         self.available_names_or_ids = set()
         self.validate = validate
         self.Model = Model
-        self.fields_get = self.Model.fields_get()
+
+        if 'lang' in Model.env.context:
+            # we don't need translations in the call fields_get() below
+            Model = Model.with_context(lang=None)
+        self.fields_get = Model.fields_get()
```

Comment:
Just do that unconditionally?
```suggestion
self.fields_get = Model.with_context(lang=None).fields_get()
```
?

And the commit message is overly verbose and providing informations we probably don't care about, I'd only keep the last two paragraphs:

> `NameManager`'s validation passes should not be language-dependent. Fetching translated model & view information thus incurs additional costs for no resaon.

> Ensure translations are never enabled for `NameManager`'s work, reducing its cost from ~13% to ~8.5% of view rendering.

Also would need some information as to e.g. which view & setup was involved here.

Reply:
`with_context` creates a new environment, it is cheap to create a new env for not free. Because we instantiate many NameManagers, we gain a few CPU cycles by only changing the context when it is needed. @Xavier-Do suggested we remove the lang from the context one for all earlier but I couldn't figure how to do so.

I'll change the commit message to include the list of modules by DB had during the benchmark.

Reply:
> `with_context` creates a new environment, it is cheap to create a new env for not free. Because we instantiate many NameManagers, we gain a few CPU cycles by only changing the context when it is needed.

If you're doing that kind of useless micro-optimisations, then you should also skip the context update when `lang` is already `None`, which you're not.

Reply:
I rerun the benchmark, `if Model.env.context.get('lang')` is a micro-optimisation compared with `with_context(lang=False)`, went for the simpler solution to always remove the lang.